moor_generator/lib/src/writer/tables/update_companion_writer.dart

```diff
@@ -200,6 +203,64 @@ class UpdateCompanionWriter {
     _buffer.write('return map; \n}\n');
   }
 
+  void _writeFromRowClass() {
+    final dataClassName = table.dartTypeName;
+
+    _buffer
+      ..write('factory ${table.getNameForCompanionClass(scope.options)}.')
+      ..write('fromRowClass($dataClassName object,')
+      ..write('{bool nullToAbsent = false}) {\n');
+
+    _buffer
+      ..write('return ')
+      ..write(table.getNameForCompanionClass(scope.options))
+      ..write('(');
+
+    final info = table.existingRowClass;
+    final named = <MoorColumn, String>{};
+
+    info.mapping.forEach((column, parameter) {
+      if (parameter.isNamed) {
+        named[column] = parameter.name;
+      }
+    });
+
+    for (final field in named.values) {
+      final column = table.columns.firstWhere(
+          (element) => element.name.name == field,
+          orElse: () => null);
+
+      if (column == null) {
+        break;
+      }
+
+      final dartName = column.dartGetterName;
+      _buffer..write(dartName)..write(': ');
+
+      final needsNullCheck = column.nullable || !scope.generationOptions.nnbd;
+      if (needsNullCheck) {
+        _buffer
+          ..write('object.$dartName')
+          ..write(' == null && nullToAbsent ? const Value.absent() : ');
+        // We'll write the non-null case afterwards
+      }
+
+      _buffer..write('Value (')..write('object.$dartName')..write('),');
+    }
+
+    _buffer.writeln(');\n}\n');
+
+    _buffer
+      ..write('static Map<String, Expression> ')
+      ..write('fromRowClassToColumns($dataClassName object,')
+      ..write('{bool nullToAbsent = false}) {\n')
+      ..write('return ${table.getNameForCompanionClass(scope.options)}.')
+      ..write('fromRowClass(object, nullToAbsent: nullToAbsent)')
+      ..write('.toColumns(nullToAbsent);');
```

Comment:
The overall mapping constructor looks very good, but do we really need this helper? I think users can just do that manually, we don't need to generate a utility function for each companion here.

Reply:
It's not really necessary. I made for to set default (false) value to nullToAbsent parameter in toColumns() function. Maybe it's not clear to user, but 'nullToAbsent' have to be false in this case. The row class is probably mutable when user want use this mapping feature. It's an unexcpected operation when user set a nullable field to null (that had a value before), and moor absents and skip update that field after mapping. In fact, maybe we must fix nullToAbsent to false.